packages/mobile/src/web3/saga.ts

```diff
@@ -36,68 +34,67 @@ const MNEMONIC_BIT_LENGTH = (ETH_PRIVATE_KEY_LENGTH * 8) / 2
 
 const TAG = 'web3/saga'
 // The timeout for web3 to complete syncing and the latestBlock to be > 0
-const CHECK_SYNC_PROGRESS_TIMEOUT = 60000
+export const SYNC_TIMEOUT = 60000
 const BLOCK_CHAIN_CORRUPTION_ERROR = "Error: CONNECTION ERROR: Couldn't connect to node on IPC."
 
-// checks if web3 claims it is currently syncing or not
-function* checkWeb3SyncProgressClaim() {
+// checks if web3 claims it is currently syncing and attempts to wait for it to complete
+export function* checkWeb3SyncProgress() {
   while (true) {
     try {
-      const syncProgress = yield web3.eth.isSyncing() // returns true when it's still syncing and thus not ready
-      if (typeof syncProgress === 'boolean' && !syncProgress) {
-        Logger.debug(TAG, 'checkWeb3SyncProgressClaim', 'sync complete')
+      Logger.debug(TAG, 'checkWeb3SyncProgress', 'Checking sync progress')
+
+      // isSyncing returns a syncProgress object when it's still syncing, false otherwise
+      const syncProgress = yield web3.eth.isSyncing()
 
-        yield put(setSyncProgress(100))
-        return true
+      if (typeof syncProgress === 'boolean' && !syncProgress) {
+        Logger.debug(TAG, 'checkWeb3SyncProgress', 'Sync maybe complete, checking')
+
+        const latestBlock: Block = yield call(getLatestBlock)
+        if (latestBlock && latestBlock.number > 0) {
+          yield put(setLatestBlockNumber(latestBlock.number))
+          Logger.debug(TAG, 'checkWeb3SyncProgress', 'Sync is complete')
+          return true
+        } else {
+          Logger.debug(TAG, 'checkWeb3SyncProgress', 'Sync not actually complete, still waiting')
+        }
+      } else {
+        yield put(updateWeb3SyncProgress(syncProgress))
       }
-      Logger.debug(TAG, 'checkWeb3SyncProgressClaim', 'sync in progress')
-      yield put(updateWeb3SyncProgress(syncProgress))
-      // not ready yet, keep looping
+
+      yield delay(100) // wait 100ms while web3 syncs
     } catch (error) {
       if (error.toString().toLowerCase() === BLOCK_CHAIN_CORRUPTION_ERROR.toLowerCase()) {
         CeloAnalytics.track(CustomEventNames.blockChainCorruption, {}, true)
         const deleted = yield call(deleteChainData)
         if (deleted) {
           navigateToError('corruptedChainDeleted')
         }
-        throw new Error('Corrupted chain data encountered')
       } else {
-        Logger.error(TAG, `checking web3 sync progress: ${error}`)
+        Logger.error(TAG, 'Unexpected sync error', error)
```

Comment:
@martinvol it's better to provide the error as the 3rd param to Logger.error, which has special logic to extract and format the error message nicely